### PR TITLE
V3 Events & Snaps

### PR DESCRIPTION
**Pull request by Serafadam**

### CODE REVIEW

File: examples/cpp/HostNodes/events.cpp [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+#include <string>
+
+#include "depthai/depthai.hpp"
+#include "depthai/utility/EventsManager.hpp"
+
+int main(int argc, char* argv[]) {
+    dai::Pipeline pipeline(true);
+
+    auto eventsManager = std::make_shared<dai::utility::EventsManager>();
+    eventsManager->setUrl("https://events-ingest.apps.stg.hubcloud");
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply examples/cpp/HostNodes/events.cpp:13]: The default URL is for production but AFAIK snaps & events are not available there yet, so for now we need to use this URL

File: include/depthai/utility/EventsManager.hpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+     * @param extraData Extra data to send
+     * @param deviceSerialNo Device serial number
+     * @return void
+     */
+    void sendSnap(const std::string& name,
+                  const std::shared_ptr<ImgFrame>& imgFrame = nullptr,
+                  std::vector<std::shared_ptr<EventData>> data = {},
+                  const std::vector<std::string>& tags = {},
+                  const std::unordered_map<std::string, std::string>& extraData = {},
+                  const std::string& deviceSerialNo = "");
```
[the remark this answers is not in the document]
Comment (Serafadam) [reply include/depthai/utility/EventsManager.hpp:71]: Error as in `bool`? Or custom error code?

[the remark this answers is not in the document]
Comment (moratom) [reply include/depthai/utility/EventsManager.hpp]: We can do bool I think.

[the remark this answers is not in the document]
Comment (moratom) [reply include/depthai/utility/EventsManager.hpp]: And keep the logger::error, now there's no feedback if the function is not successful, if one is not looking into the terminal output.

File: examples/cpp/HostNodes/events.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (moratom) [reply examples/cpp/HostNodes/events.cpp:13]: I think this is now available right?

File: recording.mp4
Comment (moratom) [inline recording.mp4:1]: Force push out of the history

File: bindings/python/src/utility/EventsManagerBindings.cpp
Comment (moratom) [inline bindings/python/src/utility/EventsManagerBindings.cpp:1]: This likely doesn't compile when protobuf support is disabled right?

**Review by moratom — CHANGES_REQUESTED** [review]
Let's make the event manager enable-able with it's own flag and just code in the logic that protobuf is required for it and test that the code can be compiled without it, otherwise LGTM

**Review by moratom — APPROVED** [review]

**Pull request merged**